go-controller/pkg/libovsdbops/template_var.go

```diff
@@ -0,0 +1,121 @@
+package libovsdbops
+
+import (
+	"context"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+type chassisTemplateVarPredicate func(*nbdb.ChassisTemplateVar) bool
+
+// ListTemplates looks up all chassis template variables.
+func ListTemplates(nbClient libovsdbclient.Client) ([]*nbdb.ChassisTemplateVar, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+
+	templatesList := []*nbdb.ChassisTemplateVar{}
+	err := nbClient.List(ctx, &templatesList)
+	return templatesList, err
+}
+
+// CreateOrUpdateChassisTemplateVarOps creates or updates the provided
+// 'template' variable and returns the corresponding ops.
+func CreateOrUpdateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.CreateOrUpdateOps(ops, operationModel{
+		Model: template,
+		ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+			return item.Chassis == template.Chassis
+		},
```

Comment:
Wait, I just checked the model and the ChassisTemplateVar.Chassis is an schema index, so we don't really need the predicate?

Reply:
True, we can skip this.